### PR TITLE
Use async FileStreams where it makes sense

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

File: MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs [hunk 1]
```diff
@@ -1557,7 +1557,7 @@ private TransportStreamTimestamp GetMpegTimestamp(string path)
         {
             var packetBuffer = new byte[197];
 
-            using (var fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read))
+            using (var fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, 1))
```
Comment (crobibero) [inline MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs:1560]: Why not use the same `bufferSize` as the `packetBuffer` length?

Comment (Bond-009) [reply MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs:1560]: That would allocate another buffer, 1 is the magic number so `FileStream` doesn't create an internal buffer.

**Comment by cvium** [comment]
Should this be blocked until .NET 6? Async filestreams in Windows is pure garbage.

**Comment by Bond-009** [comment]
We could, or I could add a temp change so it doesn't do it on Windows?

**Comment by cvium** [comment]
> 
> 
> We could, or I could add a temp change so it doesn't do it on Windows?

Either is fine with me. I just don't want Windows filestream to slow the server down

**Review by cvium — APPROVED** [review]

**Review by crobibero — APPROVED** [review]

**Pull request merged**